src/models/UnfinishedSteps.php

```diff
@@ -114,29 +114,59 @@ private function cleanUpResult(array $res): array
 
     private function getTeamWhereClause(EntityType $model): string
     {
-        $sql =  'WHERE ' . ($model === EntityType::Items ? 'entity.team = :teamid ' : '1 = 1 ');
-        // add pub/org/team filter
-        $sqlPublicOrg = sprintf("( (JSON_EXTRACT(entity.canread, '$.base') = %d OR JSON_EXTRACT(entity.canread, '$.base') = %d) AND entity.userid = users2teams.users_id) OR ", BasePermissions::Full->value, BasePermissions::Organization->value);
-        $sql .= sprintf(" AND  %s (JSON_EXTRACT(entity.canread, '$.base') = %d AND users2teams.users_id = entity.userid) OR (JSON_EXTRACT(entity.canread, '$.base') = %d ", $sqlPublicOrg, BasePermissions::MyTeams->value, BasePermissions::User->value);
+        // add pub/org
+        $sql = sprintf(
+            "%s AND (
+                (JSON_EXTRACT(entity.canread, '$.base') = %d OR JSON_EXTRACT(entity.canread, '$.base') = %d)
+                AND users2teams.users_id = entity.userid
+            )",
+            $model === EntityType::Items ? 'entity.team = :teamid' : '1 = 1',
```

Comment:
@NicolasCARPi maybe you want to carefully review if the logic in the sql reflects what is expected. In particular `$model === EntityType::Items ? 'entity.team = :teamid' : '1 = 1',` is only applied to pub/org filter but not to the other filters due to precedence of `AND` over `OR`.

Reply:
I see. I've simply modified the comment to make it more clear that this part adds the pub filter but also the team filter for items.